qualtran/bloqs/mod_arithmetic/mod_subtraction_test.py

```diff
@@ -219,7 +219,7 @@ def test_cmodsub_bloq_counts(cv, dtype, bitsize, prime):
 @pytest.mark.parametrize('cv', range(2))
 @pytest.mark.parametrize('dtype', [QUInt, QMontgomeryUInt])
 @pytest.mark.parametrize(
-    ['bitsize', 'prime'], [(p, n) for p in (13, 17, 23) for n in range(p.bit_length() - 1, 6)]
+    ['prime', 'bitsize'], [(p, n) for p in (13, 17, 23) for n in range(p.bit_length() - 1, 10)]
```

Comment:
This test has been running for more than 4 minutes and has several failures. 

Please re-consider the testing approach here. I'd suggest judiciously selecting configurations and/or randomly selecting inputs. Maybe a couple tests

 - cv=0 and a second worth of random parameters to validate that it doesn't do anything
 - cv=1; outer product of dtype; exhaustive for small bitsize
 - large bitsize random input values
 - large bitsize: hand-selected extremal values (i.e. largest possible)

Reply:
> This test has been running for more than 4 minutes and has several failures.

this should not happen ... this test passes both locally and in the CI ... the only test that was failing was a a test for ModExp which I marked with xfail since it's flaky and will look into it later.

Reply:
as for changing the test strategy I can do that

Reply:
well it won't run on the (per-PR) CI since it's marked as `slow`. Can you re-run locally to verify? I will as well

Reply:
switching "bitsize" and "prime" in the parameterize call seems wrong...

Reply:
the bug was that somehow I added -1 after p.bit_length() ... so that bitsize was too small.

fixed now and I reduced the number of tests, added a test for large primes, and added on_classical_vals method to bitwise bloqs (BitwiseNot, MultiTargetCNOT) to speedup the test

now all the tests run in `2m17.461s` on my machine

Reply:
thanks for investigating